code/modules/economy/vending_machines.dm

```diff
@@ -1080,4 +1080,43 @@
 					/obj/item/weapon/reagent_containers/food/drinks/cans/orange_cola = 1,
 					/obj/item/weapon/reagent_containers/food/drinks/cans/gingerale = 1)
 	idle_power_usage = 211 //refrigerator - believe it or not, this is actually the average power consumption of a refrigerated vending machine according to NRCan.
-	vending_sound = "machines/vending/vending_cans.ogg"
\ No newline at end of file
+	vending_sound = "machines/vending/vending_cans.ogg"
+
+///////////////////////Donk-Soft!///////////////////////////////////////
+
+/obj/machinery/vending/donksoft
+	name = "Donk-Soft!"
+	desc = "A toy vendor owned by Donk-Soft, a NanoTrasen sub-company."
+	description_fluff = "Donk-Soft is a sub-company owned by NanoTrasen that distribute replica weapons that shoot squish foam darts. They've been a staple of personal entertainment for decades but their buisness has only just moved to the fringes of the galaxy."
+	icon_state = "donksoft"
+	product_slogans = "Get your cool toys today!;Trigger a valid hunter today!;Quality toy weapons for cheap prices!;Give them to HoPs for all access!;Give them to HoS to get permabrigged!"
```

Comment:
These messages are -very- tg. Would suggest something a bit lore-friendlier here and below.